portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -2771,6 +2781,81 @@ private long _getValidFileEntryTypeId(
 		}
 	}
 
+	private void _initializeFileEntryPermissionForDLFolder(
+			DLFileEntry dlFileEntry)
+		throws PortalException {
+
+		long folderId = dlFileEntry.getFolderId();
+
+		long resourcePrimKey = folderId;
+
+		long inheritanceParentFolderId = -1;
+
+		long companyId = dlFileEntry.getCompanyId();
+
+		long groupId = dlFileEntry.getGroupId();
+
+		List<Long> ancestorFolderIds = new ArrayList<>();
+
+		if (folderId == DLFolderConstants.DEFAULT_PARENT_FOLDER_ID) {
+			inheritanceParentFolderId =
+				DLPermissionPropagationUtil.getInheritableParentFolderId(
+					companyId, groupId, ancestorFolderIds);
+
+			resourcePrimKey = groupId;
+		}
+		else {
+			DLFolder folder = dlFileEntry.getFolder();
+
+			ancestorFolderIds.add(folder.getFolderId());
+			ancestorFolderIds.addAll(folder.getAncestorFolderIds());
```

Comment:
I don't see this ancestorFolderIds variable used somewhere else. Please double check.

Reply:
Fixed it

Reply:
I fixed it